presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -112,6 +122,238 @@ When a ``GROUP BY`` clause is used in a ``SELECT`` statement all output
 expression must be either aggregate functions or columns present in
 the ``GROUP BY`` clause.
 
+**Complex Grouping Operations**
+
+Presto also supports complex aggregations using the ``GROUPING SETS``, ``CUBE``
+and ``ROLLUP`` syntax. This syntax allows users to perform analysis that require
+aggregation on multiple sets of columns in a single query. Complex grouping
+operations do not support grouping on expressions composed of input columns.
+Only column names or ordinals are allowed. All complex grouping operations can
+be specified as a ``UNION ALL`` of simple ``GROUP BY`` expressions, as shown in
+the following examples.
+
+
+**GROUPING SETS**
+
+Grouping sets allow users to specify multiple lists of columns to ``GROUP BY``
+on. The columns not part of a given sublist of ``GROUP BY`` columns are set to
+``NULL``.:
+
+    SELECT * FROM shipping;
+
+.. code-block:: none
+
+     origin_state | origin_zip | destination_state | destination_zip | package_weight
+    --------------+------------+-------------------+-----------------+----------------
+     California   |      94131 | New Jersey        |            8648 |             13
+     California   |      94131 | New Jersey        |            8540 |             42
+     New Jersey   |       7081 | Connecticut       |            6708 |            225
+     California   |      90210 | Connecticut       |            6927 |           1337
+     California   |      94131 | Colorado          |           80302 |              5
+     New York     |      10002 | New Jersey        |            8540 |              3
+    (6 rows)
+
+``GROUPING SETS`` semantics are demonstrated by this example query::
+
+    SELECT origin_state, origin_zip, destination_state, sum(package_weight)
+    FROM shipping
+    GROUP BY GROUPING SETS (
+        (origin_state),
+        (origin_state, origin_zip),
+        (destination_state));
+
+.. code-block:: none
+
+     origin_state | origin_zip | destination_state | _col0
+    --------------+------------+-------------------+-------
+     New Jersey   | NULL       | NULL              |   225
+     California   | NULL       | NULL              |  1397
+     New York     | NULL       | NULL              |     3
+     California   |      90210 | NULL              |  1337
+     California   |      94131 | NULL              |    60
+     New Jersey   |       7081 | NULL              |   225
+     New York     |      10002 | NULL              |     3
+     NULL         | NULL       | Colorado          |     5
+     NULL         | NULL       | New Jersey        |    58
+     NULL         | NULL       | Connecticut       |  1562
+    (10 rows)
+
+The preceding query may be considered logically equivalent to a ``UNION ALL`` of
```

Comment:
I suppose it's worth pointing out that `UNION ALL` does not guarantee equal translation in all cases.